annotation-dereferenciation/src/main/java/eu/europeana/annotation/dereferenciation/MetisDereferenciationClient.java

```diff
@@ -140,46 +144,72 @@ public synchronized Map<String, String> dereferenceOne(String baseUrl, String ur
      */
     public synchronized Map<String, String> dereferenceMany(String baseUrl, List<String> uris, String language) throws AnnotationDereferenciationException {
 	Map<String, String> res = new HashMap<String, String>();
-	String xmlResponse, jsonLdStr;
+	String jsonLdStr;
+	InputStream streamResponse;
 	try {
-        	@SuppressWarnings({ "rawtypes", "unchecked" })
-		String urisJson = JsonSerializer.toString((List)uris);
-	        xmlResponse = getHttpConnection().postRequest(baseUrl, urisJson);
-	        for (Object uri : uris) {
-	            jsonLdStr = convertToJsonLd((String)uri, xmlResponse, language).toString();
-        	    res.put((String)uri, jsonLdStr);
-         	}
+      	@SuppressWarnings({ "rawtypes", "unchecked" })
+      	String urisJson = JsonSerializer.toString((List)uris);
+      	streamResponse = getHttpConnection().postRequest(baseUrl, urisJson);
+      	String[] urisArray = new String[uris.size()];
+        urisArray = uris.toArray(urisArray);
+        jsonLdStr = convertToJsonLd(urisArray, streamResponse, language).toString();
+        //we need to separate the individual jsons manually
+        List<Integer> startingPositions = findSubstringIndexes(jsonLdStr, "{ \"@context\":");
```

Comment:
this looks tricky, not sure if the output is properly formatted, but we can check in an integration test. Alternatively we could use Jettison to get the nodes of the individual entities